src/tracks/playabletrack/wavetrack/ui/WaveTrackAffordanceControls.cpp

```diff
@@ -648,6 +653,18 @@ void WaveTrackAffordanceControls::OnRenderClipStretching(
    SelectInterval(project, *interval);
 }
 
+void WaveTrackAffordanceControls::OnPitchShift(AudacityProject& project, bool up)
+{
+   const auto [track, it] = SelectedIntervalOfFocusedTrack(project);
+   if (track != FindTrack().get())
+      return;
+   auto interval = *it;
+   interval->SetSemitoneShift(interval->GetSemitoneShift() + (up ? 1 : -1));
```

Comment:
This does not impose the limits on pitch shift that the dialog does.

Reply:
It's a little strange that this is a member function of a UI object (representing a certain rectangle of the screen).  Can't it simply be a non-member function?

Reply:
Looks indeed like it should.

Reply:
... there is the call to `FindTrack`, though, which is a member method. I haven't understood yet under what conditions it may cause an early return.
Making that method const, at least, is a no-brainer.

Reply:
Ok, found something better. Coming up...